### DISCUSSION TITLE
How readily complete is V?

### DISCUSSION THREAD

**Discussion by abdulbadii** [vlang/v#7808] [General]
I'm to transform/transpile Stockfish c++ in V, is it readily complete to replace all the c++ capabilities used there?
Please list whichever one seem not ready or hard to get substitute in brief clear explanation. Thanks before

**Comment by medvednikov** [comment]
It's pretty stable, the V compiler and vlib are 120k lines of V, works quite well.

You will face compiler bugs, but if you report them, they get fixed quickly ;)

**Reply by abdulbadii** [reply]
All ? I mean library for data race atomic, hash function, multi threads, zobris hash, ec etc could viably be substitute?

**Reply by fnetX** [reply]
We had a try at V in production, and we don't think it's ready yet. The compiler has some flaws which are indeed fixed immediately, but the V libraries don't feel mature yet (especially vweb) and are not very well documented. 
To use V in production, I'd probably recommend waiting another few months // until V 0.3.

For use in hobby projects, V will be a lot of fun. You'll probably be okay with the flaws mentioned above (and you always have the option to use C libraries).

**Reply by abdulbadii** [reply]
IMHO prioritized C usablity of most C static libraries/API to be callable in V is so good

**Reply by tomByrer** [reply]
There are some things in flux with V, docs for libs are bad, & not that large of an ecosystem.  However, Discord is VERY active, & V has great potential.  I would use V for small personal tools & maybe some small scripts for production, but nothing even medium-sized right now.

That said, keep following it; V is the best 'new' language out there IMHO.
